galleon-plugins/src/main/java/org/wildfly/galleon/plugin/WfInstallPlugin.java

```diff
@@ -911,4 +912,61 @@ private static void changeLineEndings(final Path file, final boolean isWindows)
             Files.delete(temp);
         }
     }
+
+    public void appendFile(FileAppender config) throws ProvisioningException {
+        if(config.getTarget() == null) {
+            if(config.isIgnore()) {
+                return;
+            }
+            throw new ProvisioningException("Target can't be null when appending content");
+        }
+        final Path target = runtime.getStagedDir().resolve(config.getTarget());
+        if (!Files.exists(target)) {
+            if(config.isIgnore()) {
+                return;
+            }
+            throw new ProvisioningException(Errors.pathDoesNotExist(target));
+        }
+        try {
+            final Path src;
+            List<String> linesToAppend;
+            if (config.getSource() != null) {
+                src = runtime.getStagedDir().resolve(config.getSource());
+                if (!Files.exists(src)) {
+                    throw new ProvisioningException(Errors.pathDoesNotExist(src));
+                }
+                linesToAppend = Files.readAllLines(src);
+            } else {
+                src = null;
+                linesToAppend = config.getLines();
+            }
+            List<String> lines = Files.readAllLines(target);
+            List<String> updatedLines = new ArrayList<>(linesToAppend.size() + 3);
+            for (String line : lines) {
+                if (config.getMatch() != null && config.getMatch().matcher(line).find()) {
+                    for (int i = 0; i < linesToAppend.size(); i++) {
+                        if (i == 0) {
+                            updatedLines.add(line + linesToAppend.get(0));
+                        } else {
+                            updatedLines.add(linesToAppend.get(i));
+                        }
+                    }
+                } else {
+                    updatedLines.add(line);
+                }
+            }
+            if (config.getMatch() == null) {
+                updatedLines.addAll(linesToAppend);
+            }
+            Path temp = Files.createTempFile(target.getFileName().toString(), "");
+            Files.write(temp, updatedLines, StandardOpenOption.WRITE);
+            Files.delete(target);
+            Files.move(temp, target);
+            if(src != null) {
+                Files.delete(src);
```

Comment:
Given that the source is provided, I would say it shouldn't be managed this way here. Let's use delete task for that.

Reply:
@jfdenise could you answer this please ?

Reply:
@aloubyansky , you mean we would need to explicitly add a delete task after the append task to remove the file? if no delete task added the file would be provisioned?

Reply:
Yes. To me as a user it would be surprising that the source of content I provided gets deleted. I know that you are the only user of this feature and that's what you need but to me this behavior as the default wouldn't be commonly expected.

Reply:
Actually I started designing it that way, with explicit delete in tasks.xml, so I am fine to revert that.
@ehsavoie you can remove the delete.